test/interval.jl

```diff
@@ -149,6 +177,58 @@ isinf(::TimeType) = false
         @test span(interval) == Hour(3)
     end
 
+    @testset "accessors - unbounded intervals" begin
+        unbounded_test_values = [
+            # one side unbounded with different types
+            (-10, nothing, 1, typeof(1), Open, Unbounded),
+            (nothing, 1, 0.01, Float64, Unbounded, Closed),
+            (nothing, 'z', 1, Char, Unbounded, Open),
+            (Date(2013, 2, 13), nothing, Day(1), Date, Closed, Unbounded),
+            (DateTime(2016, 8, 11, 0, 30), nothing, Millisecond(1), DateTime, Open, Unbounded),
+
+            #both sides unbounded different types
+            (nothing, nothing, 1, typeof(1), Unbounded, Unbounded),
+            (nothing, nothing, 0.01, Float64, Unbounded, Unbounded),
+            (nothing, nothing, 1, Char, Unbounded, Unbounded),
+            (nothing, nothing, Day(1), Day, Unbounded, Unbounded),
+            (nothing, nothing, Millisecond(1), DateTime, Unbounded, Unbounded),
+
+            # test adding eps() with unbounded
+            (-10.0, nothing, nothing, Float64, Open, Unbounded),
+            (nothing, 10.0, eps(Float64), Float64, Unbounded, Open),
+        ]
```

Comment:
May as well construct actual intervals in this case as you have all of the required information:
```suggestion
        unbounded_test_values = [
            # one side unbounded with different types
            (Interval{Open,Unbounded}(-10, nothing), 1)
            (Interval{Unbounded,Closed}(nothing, 1.0), 0.01),
            (Interval{Unbounded,Open}(nothing, 'z') 1),
            (Interval{Closed,Unbounded}(Date(2013, 2, 13), nothing), Day(1)),
            (Interval{Open,Unbounded}(DateTime(2016, 8, 11, 0, 30), nothing), Millisecond(1)),

            # both sides unbounded different types
            (Interval{Int}(nothing, nothing), 1),
            (Interval{Float64}(nothing, nothing), 0.01),
            (Interval{Char}(nothing, nothing), 1),
            (Interval{Day}(nothing, nothing), Day(1)),
            (Interval{DateTime}(nothing, nothing), Millisecond(1)),

            # test adding eps() with unbounded
            (Interval{Open,Unbounded}(-10.0, nothing), nothing),
            (Interval{Unbounded,Open}(nothing, 10.0), eps(Float64)),
        ]
```